Index: new/ksmtuned
===================================================================
--- new.orig/ksmtuned	2011-09-23 12:05:44.000000000 +0200
+++ new/ksmtuned	2011-09-23 12:06:34.000000000 +0200
@@ -72,7 +72,7 @@
     # calculate how much memory is committed to running qemu processes
     local progname
     progname=${1:-kvm}
-    ps -C "$progname" -o rsz | awk '{ sum += $1 }; END { print sum }'
+    ps -C "$progname" -o vsz= | awk '{ sum += $1 }; END { print sum }'
 }
 
 free_memory () {
